### DISCUSSION TITLE
Why is 3DES the default at "openssl req" with no option to change it?

### DISCUSSION THREAD

**Discussion by victorcruceru** [openssl/openssl#25775] [General Discussion]
Looking into the implementation of "openssl req" command I have noticed that the default encryption algorithm for private keys enveloping  is 3DES and there is no option to change it (see https://github.com/openssl/openssl/blob/master/apps/req.c line 277)
Is this 3DES still secure ?
What about providing a way to change it with other better algorithm, like AES?

**Comment by mattcaswell** [comment]
3DES is still widely used and provides reasonable security (112 bits), although NIST is retiring (has retired?) it.

IMO we should at least provide an option for replacing this cipher, and consider whether changing the default is appropriate.

**Reply by mattcaswell** [reply]
I created a feature request for this in #25776

**Comment by beldmit** [comment]
<s>DES</s>3DES is the default value also for `cms`, `smime`, and `pkcs12` commands. Shouldn't we change it?

**Reply by mattcaswell** [reply]
To be clear 3DES is - right? Perhaps we should consider changing the default for all these apps. But the immediate priority is to give the `req` command the option to change it at all.

**Reply by beldmit** [reply]
3DES, thanks, will update the comment.

I agree, `req` first, changing default cipher is a separate task

**Reply by mattcaswell** [reply]
I added an OTC hold to #25776 so we can discuss this point
